test/userdata/metrics_value_benchmark_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package metric_value_benchmark
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/environment/computetype"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+)
+
+const namespace = "UserDataTest"
+
+type MetricBenchmarkTestSuite struct {
+	suite.Suite
+	test_runner.TestSuite
+}
+
+func (suite *MetricBenchmarkTestSuite) SetupSuite() {
+	fmt.Println(">>>> Starting MetricBenchmarkTestSuite")
+}
+
+func (suite *MetricBenchmarkTestSuite) TearDownSuite() {
+	suite.Result.Print()
+	fmt.Println(">>>> Finished MetricBenchmarkTestSuite")
+}
+
+var envMetaDataStrings = &(environment.MetaDataStrings{})
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags(envMetaDataStrings)
+}
+
+var (
+	ec2TestRunners []*test_runner.TestRunner
+)
+
+func getEc2TestRunners(env *environment.MetaData) []*test_runner.TestRunner {
+	if ec2TestRunners == nil {
+		factory := dimension.GetDimensionFactory(*env)
+		ec2TestRunners = []*test_runner.TestRunner{
+			{TestRunner: &CPUTestRunner{test_runner.BaseTestRunner{DimensionFactory: factory}}},
+		}
+	}
+	return ec2TestRunners
+}
+
+func (suite *MetricBenchmarkTestSuite) TestAllInSuite() {
+	env := environment.GetEnvironmentMetaData(envMetaDataStrings)
+	switch env.ComputeType {
+	case computetype.EC2: // EC2 tests
+		log.Println("Environment compute type is EC2")
+		for _, testRunner := range getEc2TestRunners(env) {
+			if shouldRunEC2Test(env, testRunner) {
+				testRunner.Run(suite)
+			}
+		}
+	default:
+		log.Println("Invalid environment being used")
+	}
+
+	suite.Assert().Equal(status.SUCCESSFUL, suite.Result.GetStatus(), "Metric Benchmark Test Suite Failed")
+}
+
+func (suite *MetricBenchmarkTestSuite) AddToSuiteResult(r status.TestGroupResult) {
+	suite.Result.TestGroupResults = append(suite.Result.TestGroupResults, r)
+}
+
+func TestMetricValueBenchmarkSuite(t *testing.T) {
+	suite.Run(t, new(MetricBenchmarkTestSuite))
+}
+
+func shouldRunEC2Test(env *environment.MetaData, t *test_runner.TestRunner) bool {
+	if env.EC2PluginTests == nil {
+		return true // default behavior is to run all tests
+	}
+	_, ok := env.EC2PluginTests[strings.ToLower(t.TestRunner.GetTestName())]
+	return ok
+}
+
+// isAllValuesGreaterThanOrEqualToExpectedValue will compare if the given array is larger than 0
+// and check if the average value for the array is not la
+// TODO: Moving metric_value_benchmark to validator
+// https://github.com/aws/amazon-cloudwatch-agent-test/pull/162
+func isAllValuesGreaterThanOrEqualToExpectedValue(metricName string, values []float64, expectedValue float64) bool {
```

Comment:
We should not be duplicating this from the other file. Please pull this function out into a common place so we can use it everywhere.

Reply:
There are a couple PRs that do this so I won't block on it here